### PR TITLE
Add Deploy Mode to usage stats.

### PR DESCRIPTION
**Pull request by captncraig**
With 0.38.0 we added a deploy mode to the user agent, and added ways to set it, but we did not add it to the usage stats, which would be very useful. This adds it.

Also makes a best-effort attempt to detect homebrew installs, by looking t the executable path.

### CODE REVIEW

**Review by jcreixell — COMMENTED** [review]
could you please add a test [here](https://github.com/grafana/agent/blob/main/internal/useragent/useragent_test.go)? other than that, LGTM

**Review by rfratto — COMMENTED** [review]
You'll need to update this page: https://grafana.com/docs/agent/latest/data-collection/

And add this in the CHANGELOG, since we promised we'd do that any time list of collected data changes.

**Comment by captncraig** [comment]
Added tests and transparency.

File: docs/sources/data-collection.md
Comment (clayton-cornell) [inline docs/sources/data-collection.md:33]:
```suggestion
* Method used to deploy Grafana Agent, for example Docker, Helm, RPM, or Operator.
```

**Review by rfratto — APPROVED** [review]
LGTM once Clayton approves docs

**Pull request merged**
